### ISSUE TITLE
Cannot find pid file

### ISSUE THREAD

**Issue by Syer10** [ajenti/ajenti#1247]
ajenti.service - Ajenti panel
     Loaded: loaded (/lib/systemd/system/ajenti.service; enabled; vendor preset: enabled)
     Active: failed (Result: exit-code) since Fri 2020-05-01 03:18:04 UTC; 5s ago
    Process: 1434 ExecStart=/usr/bin/python3 /usr/local/bin/ajenti-panel -d (code=exited, status=0/SUCCESS)
   Main PID: 1437 (code=exited, status=1/FAILURE)

May 01 03:18:02 syer-sever systemd[1]: Starting Ajenti panel...
May 01 03:18:02 syer-sever systemd[1]: ajenti.service: Can't open PID file /run/ajenti.pid (yet?) after start: Operation not permitted
May 01 03:18:03 syer-sever systemd[1]: Started Ajenti panel.
May 01 03:18:04 syer-sever systemd[1]: ajenti.service: Main process exited, code=exited, status=1/FAILURE
May 01 03:18:04 syer-sever systemd[1]: ajenti.service: Failed with result 'exit-code'.

This is on Ubuntu 20.04 with the script provided here: 
http://docs.ajenti.org/en/latest/man/install.html

**Comment by kiarn** [comment]
Hello @Syer10 

Is there any error in `/var/log/ajenti/ajenti.log` ?

Here a updated version of the install process : https://github.com/kiarn/ajenti/blob/docs/docs/source/man/install.rst
https://github.com/kiarn/ajenti/blob/docs/scripts/install.sh

Not published yet.

**Comment by Syer10** [comment]
Just this
2020-05-01 02:57:01,217 ERROR   : Fatal crash occured
2020-05-01 02:59:01,767 ERROR   : Fatal crash occured
2020-05-01 03:01:39,229 ERROR   : Fatal crash occured
2020-05-01 03:13:16,313 ERROR   : Fatal crash occured
2020-05-01 03:15:10,696 ERROR   : Fatal crash occured
2020-05-01 03:18:03,935 ERROR   : Fatal crash occured

**Comment by Syer10** [comment]
I ran your script, same thing

**Comment by kiarn** [comment]
Ok, it's because the requirements are not included in the packages, this is solved for the next version, coming soon.

**Comment by kiarn** [comment]
Ajenti 2.1.35 is out, and I was able to install it via automated script on a fresh Ubuntu 20.04, please try it.

**Comment by Syer10** [comment]
I think its because I already have Ajenti installed, that the new version of the script isnt working for me

**Comment by benmassaoud** [comment]
Same problem here:
 23:26 ajenti.service: Failed with result 'signal'. 
 23:26 ajenti.service: Main process exited, code=killed, status=11/SEGV 
 23:26 Started Ajenti panel. 
 23:26 ajenti.service: Can't open PID file /run/ajenti.pid (yet?) after start: No such file or directory 
 23:26 Starting Ajenti panel... 

I tried in Centos 8, Ubuntu 20.04 and debian 10, same probleme, ajenti service not strated

**Comment by Trauni** [comment]
Same here

**Comment by StefanoFrazzetto** [comment]
**tl;dr I solved the issue my manually downgrading greenlet**
```
sudo python3 -m pip install greenlet==0.4.16
```
Although this fixes the issue, I would suggest moving away from Ajenti to something more reliable.
I found https://github.com/webmin/webmin to be a good software for my needs.
\
\
This is how I found the issue on Ubuntu 20.04 on Raspberry Pi

After trying to start the service through systemctl, I got
```
ubuntu systemd[1]: ajenti.service: Can't open PID file /run/ajenti.pid (yet?) after start: Operation not permitted
```

I started the process manually with `/bin/python3 /usr/local/bin/ajenti-panel -v`, which printed
```
RuntimeWarning: greenlet.greenlet size changed, may indicate binary incompatibility.
```
while the log file `/var/log/ajenti/ajenti.log` did not contain anything useful
```
2020-11-08 18:41:19,450 DEBUG   : Registering [aj.api.http.HttpMiddleware] (interface)
2020-11-08 18:41:19,451 DEBUG   : Registering [aj.api.http.HttpPlugin] (interface)
2020-11-08 18:41:19,452 DEBUG   : Registering [aj.api.http.SocketEndpoint] (interface)
2020-11-08 18:41:19,633 DEBUG   : Registering [aj.security.verifier.ClientCertificateVerificator] (service)
2020-11-08 18:41:19,634 DEBUG   : Registering [aj.auth.AuthenticationMiddleware] (service)
2020-11-08 18:41:19,634 DEBUG   : Registering [aj.auth.AuthenticationProvider] (interface)
2020-11-08 18:41:19,634 DEBUG   : Registering [aj.auth.OSAuthenticationProvider] (implementation of [aj.auth.AuthenticationProvider])
2020-11-08 18:41:19,635 DEBUG   : Registering [aj.auth.AuthenticationService] (service)
2020-11-08 18:41:19,635 DEBUG   : Registering [aj.auth.PermissionProvider] (interface)
2020-11-08 18:41:19,637 DEBUG   : Registering [aj.routing.CentralDispatcher] (service)
2020-11-08 18:41:19,639 DEBUG   : Registering [aj.gate.middleware.SocketIORouteHandler] (service)
2020-11-08 18:41:19,640 DEBUG   : Registering [aj.gate.middleware.GateMiddleware] (service)
2020-11-08 18:41:20,876 DEBUG   : Popen: ('lsb_release', '-a')
```

After doing some research, I stumbled on https://github.com/gevent/gevent/issues/1260, and after checking the packages versions with `python3 -m pip freeze` I noticed the package was `greenlet==0.4.17`, so I downgraded it to v0.4.16, which fixed the issue.

I guess the a new pip package has not been published after https://github.com/ajenti/ajenti/commit/bbce12fd167d3fa3f5d29244731eb79d69ef24d9. 

I still get the error "Operation not permitted", but the service starts without other problems (so far)
```
Nov 08 19:04:17 ubuntu systemd[1]: Starting Ajenti panel...
Nov 08 19:04:17 ubuntu systemd[1]: ajenti.service: Can't open PID file /run/ajenti.pid (yet?) after start: Operation not permitted
Nov 08 19:04:23 ubuntu systemd[1]: Started Ajenti panel.
```

**Comment by cabusar** [comment]
Same problem on a fresh install of Debian 10, the command below solved the issue, thank you Stefano.

`sudo python3 -m pip install greenlet==0.4.16`

Ajenti is a nice product thank to the dev. ;)
